test/orcv2.jl

```diff
@@ -0,0 +1,235 @@
+
+        jd = JITDylib(es, "my.so")
+        jd_bare = JITDylib(es, "mybare.so", bare=true)
+
+        @test lookup(es, "my.so") === jd
+
+        jd_main = JITDylib(lljit)
+    end
+end
+
+@testset "Undefined Symbol" begin
+    LLJIT() do lljit
+        @test_throws LLVMException lookup(lljit, string(gensym()))
+    end
+
+    LLJIT(;tm=JITTargetMachine()) do lljit
+        jd = JITDylib(lljit)
+
+        TSContext() do ts_ctx
+            ctx = context(ts_ctx)
+            mod = LLVM.Module("jit"; ctx)
+
+            T_Int32 = LLVM.Int32Type(ctx)
+            ft = LLVM.FunctionType(T_Int32, [T_Int32, T_Int32])
+            fn = LLVM.Function(mod, "mysum", ft)
+            linkage!(fn, LLVM.API.LLVMExternalLinkage)
+
+            fname = String(mangle(lljit, "wrapper"))
+            wrapper = LLVM.Function(mod, fname, ft)
+            # generate IR
+            Builder(ctx) do builder
+                entry = BasicBlock(wrapper, "entry"; ctx)
+                position!(builder, entry)
+
+                tmp = call!(builder, fn, [parameters(wrapper)...])
+                ret!(builder, tmp)
+            end
+
+            # TODO: Get TM from lljit?
+            tm = JITTargetMachine()
+            triple!(mod, triple(lljit))
+            ModulePassManager() do pm
+                add_library_info!(pm, triple(mod))
+                add_transform_info!(pm, tm)
+                run!(pm, mod)
+            end
+            verify(mod)
+
+            ts_mod = TSModule(mod; ctx=ts_ctx)
+
+            add!(lljit, jd, ts_mod)
+            @test_throws LLVMException lookup(lljit, fname)
+        end
+    end
+end
+
+@testset "Loading ObjectFile" begin
+    LLJIT(;tm=JITTargetMachine()) do lljit
+        jd = JITDylib(lljit)
+
+        TSContext() do ts_ctx
+            ctx = context(ts_ctx)
+            sym = String(mangle(lljit, "SomeFunction"))
+
+            mod = LLVM.Module("jit"; ctx)
+            ft = LLVM.FunctionType(LLVM.VoidType(ctx))
+            fn = LLVM.Function(mod, sym, ft)
+
+            Builder(ctx) do builder
+                entry = BasicBlock(fn, "entry"; ctx)
+                position!(builder, entry)
+                ret!(builder)
+            end
+            verify(mod)
+
+            tm  = JITTargetMachine()
+            obj = emit(tm, mod, LLVM.API.LLVMObjectFile)
+            add!(lljit, jd, MemoryBuffer(obj))
+
+            addr = lookup(lljit, sym)
+
+            @test addr != 0
+
+            empty!(jd)
+            @test_throws LLVMException lookup(lljit, sym)
+        end
+    end
+end
+
+@testset "ObjectLinkingLayer" begin
+    called = Ref{Int}(0)
+
+    function ObjectLinkingLayerCreator(ctx::Ptr{Cvoid}, es::LLVM.API.LLVMOrcExecutionSessionRef, triple::Ptr{Cchar})
+        es = ExecutionSession(es)
+        oll = ObjectLinkingLayer(es)
+
+        register!(oll, GDBRegistrationListener())
+
+        called[] += 1
+
+        return oll.ref
+    end
+
+    let callback = @cfunction($ObjectLinkingLayerCreator,
```

Comment:
conditionalize these tests on platforms that have trampoline support, or use a global.